omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,180 @@
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/visitconfiguration")
+public class VisitConfigurationController2_0 extends BaseRestController {
+
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	@ResponseBody
+	public Object getCurrentConfiguration() {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		EncounterService encounterService = Context.getEncounterService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		VisitConfiguration visitConfiguration = new VisitConfiguration();
+		visitConfiguration.setEnableVisits(getEnableVisitsValue(administrationService));
+		visitConfiguration
+				.setEncounterVisitsAssignmentHandler(getVisitEncounterHandlerValue(administrationService, encounterService));
+		visitConfiguration.setStartAutoCloseVisitsTask(getAutoCloseVisitsTaskStartedValue(schedulerService));
+		visitConfiguration.setVisitTypesToAutoClose(getVisitTypesToAutoCloseValue(administrationService, visitService));
+
+		return ConversionUtil.convertToRepresentation(visitConfiguration, Representation.FULL);
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
+	@ResponseStatus(HttpStatus.OK)
+	public void updateCurrentConfiguration(@RequestBody VisitConfiguration newConfiguration) throws SchedulerException {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		// validate
+		if (newConfiguration.getEnableVisits() && StringUtils.isEmpty(newConfiguration.getEncounterVisitsAssignmentHandler())) {
+			throw new IllegalRequestException("Encounter Visit assignment handler cannot be empty");
+		}
+
+		updateEnableVisitsValue(administrationService, newConfiguration.getEnableVisits());
+
+		if (newConfiguration.getEnableVisits()) {
+			administrationService.setGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER,
+					newConfiguration.getEncounterVisitsAssignmentHandler());
+		}
+		updateGetAutoCloseVisitsTaskStartedValue(schedulerService, newConfiguration.getStartAutoCloseVisitsTask());
+		updateVisitTypesToAutoCloseValue(administrationService, visitService, newConfiguration.getVisitTypesToAutoClose());
+	}
+
+	private Boolean getEnableVisitsValue(AdministrationService administrationService) {
+		String enableVisits = administrationService
+				.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, Boolean.FALSE.toString());
+		return Boolean.parseBoolean(enableVisits);
+	}
+
+	private void updateEnableVisitsValue(AdministrationService administrationService, Boolean value) {
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS,
```

Comment:
Do we really need this method?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/d3697e81f2f6c0f95fa2a34294c2310f549264b1